modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java

```diff
@@ -0,0 +1,228 @@
+                        keyPositionMap);
+        Assert.assertNotNull(keyValues, "Access token Key value is null.");
+        accessToken = keyValues.get(0).getValue();
+
+        EntityUtils.consume(response.getEntity());
+        Assert.assertNotNull(accessToken, "Access token is null.");
+    }
+
+    @Test(groups = "wso2.is", description = "Test introspection endpoint", dependsOnMethods = "testGetAccessToken")
+    public void testIntrospectionEndpoint() throws Exception {
+
+        String scope = getScopesFromIntrospectionResponse();
+        doTheScopeValidationBasedOnTheTestUserMode(scope);
+    }
+
+    @Test(groups = "wso2.is", description = "Test introspection endpoint", dependsOnMethods = "testIntrospectionEndpoint")
+    public void getTokenAndValidate() throws Exception {
+
+        client = HttpClientBuilder.create().disableRedirectHandling().build();
+
+        try {
+            Secret password = new Secret(userPassword);
+            AuthorizationGrant passwordGrant = new ResourceOwnerPasswordCredentialsGrant(username, password);
+            ClientID clientID = new ClientID(consumerKey);
+            Secret clientSecret = new Secret(consumerSecret);
+            ClientAuthentication clientAuth = new ClientSecretBasic(clientID, clientSecret);
+            URI tokenEndpoint = new URI(OAuth2Constant.ACCESS_TOKEN_ENDPOINT);
+            Scope systemScope = new Scope(SYSTEM_SCOPE);
+            TokenRequest request = new TokenRequest(tokenEndpoint, clientAuth, passwordGrant, systemScope);
+
+            HTTPResponse tokenHTTPResp = request.toHTTPRequest().send();
+            Assert.assertNotNull(tokenHTTPResp, "Access token http response is null.");
+            AccessTokenResponse tokenResponse = AccessTokenResponse.parse(tokenHTTPResp);
+            Assert.assertNotNull(tokenResponse, "Access token response is null.");
+            accessToken = tokenResponse.getTokens().getAccessToken().getValue();
+            String scope = getScopesFromIntrospectionResponse();
+            doTheScopeValidationBasedOnTheTestUserMode(scope);
+        } finally {
+            client.close();
+        }
+    }
+
+    private String getScopesFromIntrospectionResponse() throws Exception {
+
+        String introspectionUrl = "carbon.super".equalsIgnoreCase(activeTenant) ?
+                OAuth2Constant.INTRO_SPEC_ENDPOINT : OAuth2Constant.TENANT_INTRO_SPEC_ENDPOINT;
+        org.json.simple.JSONObject introspectionResponse =
+                introspectTokenWithTenant(client, accessToken, introspectionUrl, username, userPassword);
+        Assert.assertTrue(introspectionResponse.containsKey("scope"));
+        return introspectionResponse.get("scope").toString();
+    }
+
+    private void doTheScopeValidationBasedOnTheTestUserMode(String scope) {
+
+        if (testUserMode == TestUserMode.SUPER_TENANT_ADMIN) {
+            Assert.assertTrue(scope.contains("internal_server_admin"), "Scope should contain " +
```

Comment:
Don't we have enums/constants for these "internal_server_admin" etc ?

Reply:
Could not find any constants for these as well